front/components/ConnectorPermissionsTree.tsx

```diff
@@ -85,14 +86,18 @@ export function PermissionTreeChildren({
   // If the user pressed "select all", we want to display "unselect all" and vice versa.
   // But if the user types in the search bar, we want to reset the button to "select all".
   const [selectAllClicked, setSelectAllClicked] = useState(false);
-
   const { resources, isResourcesLoading, isResourcesError } =
     useConnectorPermissionsHook({
       owner,
       dataSource,
       parentId,
       filterPermission: permissionFilter || null,
     });
+  const { parentsById } = useParentResourcesById({
+    owner,
+    dataSource,
+    selectedResources: resources,
+  });
```

Comment:
It might be more useful to get the parents of all selected resources , not of the currenlty displayed resources. 
The selected resources can be retrieved with : 
```
  const { resources: selectedResources } = useConnectorPermissionsHook({
    owner,
    dataSource,
    parentId: null,
    filterPermission: "read",
  });
```
Then, trying to get the parents with useParentResourcesById.
Need to be done only once for the whole tree (so not in PermissionTreeChildren, more in the parent component ).